packages/jest-worker/src/types.ts

```diff
@@ -56,6 +59,10 @@ export type PoolExitResult = {
   forceExited: boolean;
 };
 
+export interface PromiseWithCustomMessage<T> extends Promise<T> {
+  onCustomMessage?: (listener: OnCustomMessage) => () => void;
```

Comment:
this feels slightly wonky, but if it works I'm all for it. maybe

```suggestion
  unstable_onCustomMessage?: (listener: OnCustomMessage) => () => void;
```

just so people know not to use it if they find it? Leaves us with some wiggle-room